### PR TITLE
Updates for network isolation and account key removal

### PR DESCRIPTION
**Pull request by TomAugspurger**
Updates to the deployment and runtime to

1. Work with storage accounts where account key access is disabled
2. Comply with stricter network isolation requirements

### CODE REVIEW

File: deployment/terraform/resources/batch.tf [hunk 1]
```diff
@@ -5,6 +5,9 @@ resource "azurerm_storage_account" "pctasks-batch" {
   account_tier                    = "Standard"
   account_replication_type        = "LRS"
   allow_nested_items_to_be_public = false
+  
+  # Shared access key gets disabled via deploy script
+  # shared_access_key_enabled       = false
```
Comment (TomAugspurger) [inline deployment/terraform/resources/batch.tf:10]: Should this be uncommented?

File: pctasks/run/pctasks/run/argo/client.py [hunk 2]
```diff
@@ -130,33 +130,40 @@ def submit_workflow(
         executor_config: WorkflowExecutorConfig,
         runner_image: str,
     ) -> Dict[str, Any]:
+        print("=== SUBMITTING WORKFLOW ===")
```
Comment (TomAugspurger) [inline pctasks/run/pctasks/run/argo/client.py:133]: OK to remove all these prints?

File: pctasks/run/pctasks/run/task/prepare.py [hunk 3]
```diff
@@ -174,9 +174,11 @@ def write_task_run_msg(run_msg: TaskRunMessage, settings: RunSettings) -> BlobCo
         run_msg.encoded(),
     )
 
+    user_delegation_key = get_user_delegation_key(settings.blob_account_url)
     input_blob_sas_token = generate_blob_sas(
         account_name=settings.blob_account_name,
-        account_key=settings.blob_account_key,
+        user_delegation_key=user_delegation_key,
+        # account_key=settings.blob_account_key,
```
Comment (TomAugspurger) [inline pctasks/run/pctasks/run/task/prepare.py:181]: Can delete these.

File: scripts/publish [hunk 4]
```diff
@@ -78,10 +78,10 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
         if [ "${ONLY_IMAGE}" ]; then
             publish ${ONLY_IMAGE}
         else
-            publish pctasks-ingest
             publish pctasks-server
             publish pctasks-run
-            publish pctasks-task-base
+            # publish pctasks-ingest
```
Comment (TomAugspurger) [inline scripts/publish:83]: Uncomment these?

File: deployment/terraform/resources/batch.tf (hunk 1 above)
Comment (mmcfarland) [reply deployment/terraform/resources/batch.tf:10]: No, but it can be deleted. The deploy script will set that on/off and not have it managed by tf.

**Review by TomAugspurger — COMMENTED** [review]
I plan to merge this once CI passes.

File: pctasks/run/pctasks/run/argo/client.py (hunk 2 above)
Comment (mmcfarland) [reply pctasks/run/pctasks/run/argo/client.py]: Yeah, all the `prints` were just to help in my remote print statement debugging, feel free to delete any/all.

**Review by mmcfarland — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Test failures hopefully fixed (azurite doesn't support token credentials, so we need to use the account key in `from_account_key`, rather than generating a SAS token)

**Comment by TomAugspurger** [comment]
New failure at https://github.com/microsoft/planetary-computer-tasks/actions/runs/9863208116/job/27235465737?pr=309#step:6:78.

```
=================================== FAILURES ===================================
__________________________ test_walk_match_full_path ___________________________

    def test_walk_match_full_path():
        storage = LocalStorage(HERE / ".." / "data-files" / "simple-assets")
        subdirs = {
            root: files
            for root, _, files in storage.walk(
                min_depth=1, max_depth=1, matches="a/asset-.*.json", match_full_path=True
            )
        }
>       assert subdirs["a"] == ["asset-a-1.json", "asset-a-2.json"]
E       AssertionError: assert ['asset-a-2.j...set-a-1.json'] == ['asset-a-1.j...set-a-2.json']
E         At index 0 diff: 'asset-a-2.json' != 'asset-a-1.json'
E         Use -v to get the full diff
```

I couldn't reproduce it. Updated the test to ignore the order...

**Comment by TomAugspurger** [comment]
I don't know why the integration tests are failing (presumably auth?) but AFAICT there's nothing in the logs indicating what the issue is.

I haven't been able to run the integration tests locally. I'm inclined to just skip them at this point :/

**Comment by TomAugspurger** [comment]
I think we'll need to revert some of the changes to the Helm chart removing the account key from the pctasks-server Deployment. IIUC these integration tests are using Azurite, which requires the account key. I'll try that quick.

**Comment by TomAugspurger** [comment]
Wohoo, finally passing.

**Pull request merged**